src/device/tester.rs

```diff
@@ -0,0 +1,67 @@
+use core::sync::atomic::{AtomicUsize, Ordering};
+
+use crate::device::{DeviceAccess, Width};
+
+// ————————————————————————————— Virtual Test Device —————————————————————————————— //
+
+pub const TEST_DEVICE_SIZE: usize = 0x8;
+
+#[derive(Debug)]
+pub struct VirtTestDevice {
```

Comment:
Maybe we can have a small doc comment here to explain what the purpose of this device is (testing).